EncodingStreamingHighAvailability/HighAvailability.JobVerification/JobVerificationFunction.cs

```diff
@@ -0,0 +1,66 @@
+namespace HighAvailability.JobVerification
+{
+    using HighAvailability.Helpers;
+    using HighAvailability.Interfaces;
+    using HighAvailability.Models;
+    using Microsoft.Azure.WebJobs;
+    using Microsoft.Extensions.Logging;
+    using Newtonsoft.Json;
+    using System;
+    using System.Threading.Tasks;
+
+    /// <summary>
+    /// Implements job verification Azure function. It is triggered by messages in job-verification-requests Azure queue
+    /// </summary>
+    public class JobVerificationFunction
+    {
+        /// <summary>
+        /// Json settings to deserialize data using full type names. 
+        /// </summary>
+        private static readonly JsonSerializerSettings jsonSettings = new JsonSerializerSettings { TypeNameHandling = TypeNameHandling.All };
+
+        /// <summary>
+        /// Service to process job verification requests
+        /// </summary>
+        private IJobVerificationService jobVerificationService { get; set; }
+
+        /// <summary>
+        /// Constructor
+        /// </summary>
+        /// <param name="jobVerificationService">Service to process job verification requests</param>
+        public JobVerificationFunction(IJobVerificationService jobVerificationService)
+        {
+            this.jobVerificationService = jobVerificationService ?? throw new ArgumentNullException(nameof(jobVerificationService));
+        }
+
+        /// <summary>
+        /// This function is triggered by messages in job-verification-requests Azure queue.
+        /// It runs job verification logic to verify that jobs are completed successfully or resubmit failed ones.
+        /// If function fails, exception is thrown and message is automatically reprocessed up to 5 times by default.
```

Comment:
can that 5 x be configured someplace? If so where?

Reply:
Is there exponential backoff here?

Reply:
This is out of the box Azure Function behavior that uses Azure Queue trigger. It does not have exponential backoff here. We can set visibility timeout how long to wait till next try to process message. More info is here https://docs.microsoft.com/en-us/azure/azure-functions/functions-bindings-storage-queue-output?tabs=csharp#hostjson-settings I will add this link to the documenation.